### PR TITLE
refactor(layouts): remove additional function to register backend lay…

### LINKED ISSUE #166: remove registerBackendLayouts function
we can move this function https://github.com/TYPO3-Initiatives/nuxt-typo3/blob/master/lib/templates/middleware/context.js#L24 
to `BeDynamic`
 then we can remove it 
https://github.com/TYPO3-Initiatives/nuxt-typo3/blob/master/lib/templates/plugins/layouts.js#L5 
Now we check `context.app.components` too fast, so thats why we have to use `registerBackendLayouts` to register backend layout components - it was useful only for checking and confirming that the backend layout which we want to load exist in components array.

### PR DESCRIPTION
**Pull request by mercs600**
…outs (#166)

resolve #166

### CODE REVIEW

**Comment by mercs600** [comment]
@gdulamat last part of breakings and we can release 1.0.0, please notice - I have simplified updates on nuxt context.

File: docs/frontend/backend-layouts.md [hunk 1]
```diff
@@ -6,21 +6,25 @@ In the backednd layout you can define multiple columns where you want to put you
...
+  "appearance": {
+    "layout": "layout-0",
+    "backendLayout": "2Columns"
+  },
   ...
 }
 ```
 
-If the response is ok, we can create component file. 
+If the response is ok - we see there are 2 object in content and custom backendLayout, then we can create layout file. To render content from selected `colPos` you can use `T3Renderer` .
```
Comment (gdulamat) [inline docs/frontend/backend-layouts.md:25]: We can see 2 objects - `content` and `appearance` with `backendLayout` property.

If the response is ok, we will receive 2 objects - `content` and `appearance`. The name of the backend layout is stored in the `appearance` object as `backendLayout` property, so we should create a layout file named after data provided by API.
To render content from selected `colPos` you can use `T3Renderer`.

File: docs/frontend/backend-layouts.md [hunk 2]
```diff
@@ -39,38 +43,28 @@ export default {
...
-
 ```
 
+
 ## Register backend layouts
 
-You have to register component layouts to Nuxt application context - this is needed to provide the backend layout for SSR purposes. Just use `registerBackendLayouts` plugin
-   
-create: `plugins/layouts.js`:
+You can register backend layout as a regular global Vue component.
```
Comment (gdulamat) [inline docs/frontend/backend-layouts.md:51]: I think `have to` (like before) was more accurate.

File: docs/frontend/backend-layouts.md [hunk 3]
```diff
@@ -6,21 +6,25 @@ In the backednd layout you can define multiple columns where you want to put you
...
+  },
   ...
 }
 ```
 
-If the response is ok, we can create component file. 
+If the response is ok - we see there are 2 object in content and custom backendLayout, then we can create layout file. To render content from selected `colPos` you can use `T3Renderer` .
 
-create: `components/T3Bl2Columns.vue`:
+create: `components/Bl2Columns.vue`:
```
Comment (gdulamat) [inline docs/frontend/backend-layouts.md:27]: You skipped `T3` prefix ;)

File: docs/frontend/content-elements.md [hunk 4]
```diff
@@ -91,17 +91,12 @@ export default {
    
 ```js
 import Vue from 'vue'
-import CeKeyvisual as T3CeKeyvisual from '~/components/CeKeyvisual'
+import CeKeyvisual from '~/components/CeKeyvisual'
 
+Vue.component('T3CeKeyvisual', CeKeyVisual)
 const components = {
```
Comment (gdulamat) [inline docs/frontend/content-elements.md:97]: If you removed the loop of the registrations this object is redundant.

File: docs/frontend/backend-layouts.md [hunk 5]
```diff
@@ -39,38 +43,28 @@ export default {
...
-}
-
-export default ({ app }) => {
-  Vue.use(registerBackendLayouts, {
-    context: app,
-    layouts
-  })
-}
+import Bl2Columns from '~/components/Bl2Columns'
+Vue.component('T3Bl2Columns', Bl2Columns)
```
Comment (gdulamat) [inline docs/frontend/backend-layouts.md:58]: The naming of the component and file should be consistent. Maybe we should remove `T3`? Let's discuss.

File: lib/mixins/page/default.js [hunk 6]
```diff
@@ -7,11 +7,10 @@ export default {
   head () {
     return this.getMeta()
   },
-  asyncData ({ pageContent, backendLayout }) {
-    if (pageContent) {
+  asyncData ({ t3page }) {
```
Comment (gdulamat) [inline lib/mixins/page/default.js:10]: Amazing! I like this naming convention 🚀

File: docs/frontend/backend-layouts.md (hunk 5 above)
Comment (gdulamat) [reply docs/frontend/backend-layouts.md]: I changed my mind, this is the best option. You shouldn't name non-native typo3 components like this.

Comment (mercs600) [reply docs/frontend/backend-layouts.md:61]: let's stay with prefixes only for registration, T3 is needed for overriding nuxt-typo3 components, the user doesn't have to include prefixes in their own components.

**Pull request merged**